internal/indexcoord/index_coord.go

```diff
@@ -816,7 +818,7 @@ func (i *IndexCoord) assignTaskLoop() {
 					continue
 				}
 				log.Debug("The version of the task has been updated", zap.Int64("indexBuildID", indexBuildID))
-				nodeID, builderClient := i.nodeManager.PeekClient()
+				nodeID, builderClient := i.nodeManager.PeekClient(meta.indexMeta.IndexSize * 4)
```

Comment:
It is best to change to a form that can be configured with different policies. Please refer to datacoord's approach

Reply:
No need to store IndexSize (estimated) here.
Please add a SerializeSize member to store the memory amount of serialized object.

Reply:
Yeah，I added the policies for peeking client.